cub/agent/single_pass_scan_operators.cuh

```diff
@@ -637,11 +682,12 @@ struct ReduceByKeyScanTileState<ValueT, KeyT, true>
 //        value.key   = tile_descriptor.key;
 
         TileDescriptor tile_descriptor;
+
         do
         {
-            __threadfence_block(); // prevent hoisting loads from loop
-            TxnWord alias = ThreadLoad<LOAD_CG>(d_tile_descriptors + TILE_STATUS_PADDING + tile_idx);
-            tile_descriptor = reinterpret_cast<TileDescriptor&>(alias);
+          detail::delay_on_dc_gpu_or_prevent_hoisting();
```

Comment:
That's intentional. Presense of `__threadfence_block()` for pre-Pascal is functional. For Volta+ it's an optimization. Performance evaluation showed that it's better the way it is now, in other words, having a delay for 80 and nothing for Volta+ otherwise for reduce by key operator. For other operators it's not the case.

Reply:
Thanks, that's what I wanted to confirm